pyartcd/pyartcd/pipelines/build_microshift.py

```diff
@@ -56,36 +58,80 @@ def __init__(self, runtime: Runtime, group: str, assembly: str,
+                if "x86_64" not in payload_infos or "aarch64" not in payload_infos:
+                    raise ValueError("x86_64 payload and aarch64 payload are required for rebasing microshift.")
+                major, minor = util.isolate_major_minor_in_group(self.group)
+                for info in payload_infos.values():
+                    payload_version = VersionInfo.parse(info["version"])
+                    if (payload_version.major, payload_version.minor) != (major, minor):
+                        raise ValueError(f"Specified payload {info['pullspec']} does not match group major.minor {major}.{minor}: {payload_version}")
+                release_name = payload_infos["x86_64"]["version"]  # use the version of the x86_64 payload to generate the rpm version-release.
+                custom_payloads = payload_infos
+            else:
+                # rebase against named releases
+                if self.payloads:
+                    raise ValueError(f"Specifying payloads for assembly type {assembly_type.value} is not allowed.")
+                release_name = get_release_name(assembly_type, self.group, self.assembly, None)
 
             # Rebases and builds microshift
+            if assembly_type is not assembly_type.STREAM:
+                slack_client = self.runtime.new_slack_client()
+                slack_client.bind_channel(self.group)
+                slack_response = await slack_client.say(f":construction: Build microshift for assembly {self.assembly} :construction:")
+                slack_thread = slack_response["message"]["ts"]
             version, release = self.generate_microshift_version_release(release_name)
-            nvrs = await self._rebase_and_build_rpm(version, release)
+            nvrs = await self._rebase_and_build_rpm(version, release, custom_payloads)
 
             # Create a PR to pin microshift build
-            pr = await self._create_or_update_pull_request(nvrs)
+            pr = None
+            assembly_basis = get_assembly_basis(releases_config, self.assembly)
+            if assembly_basis.brew_event:
+                pr = await self._create_or_update_pull_request(nvrs)
 
             # Sends a slack message
             message = f"Hi @release-artists , microshift for assembly {self.assembly} has been successfully built."
             if pr:
                 message += f"\nA PR to update the assembly definition has been created/updated: {pr.html_url}"
-            await self._slack_client.say(message, slack_thread)
+            if slack_client:
+                await slack_client.say(message, slack_thread)
         except Exception as err:
             error_message = f"Error building microshift: {err}\n {traceback.format_exc()}"
             self._logger.error(error_message)
-            await self._slack_client.say(error_message, slack_thread)
+            if slack_client:
+                await slack_client.say(error_message, slack_thread)
```

Comment:
I see the fail notification now, do we plan on getting the retry mechanism?

Reply:
If we need a retry mechanism, i would like to get it in the slack_client class in a separate PR.

Reply:
I think you mentioned doozer retries build, so we're good